presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1679,6 +1680,42 @@ protected Scope visitExcept(Except node, Optional<Scope> scope)
             return visitSetOperation(node, scope);
         }
 
+        private boolean checkIfRelationUsedInExpressionJointlyWithOtherRelation(Expression expression, Relation relation)
```

Comment:
Simplify the name a bit - to something like: onConditionReferencesUnrelatedFields